# 温度转换

In [1]:
TempStr = input("请输入带有符号的温度值：")
if TempStr[-1] in ['f','F']:
    C = (eval(TempStr[0:-1]) - 32) / 1.8
    print("转换后的温度是{:.2f}C".format(C))
elif TempStr[-1] in ['c',"C"]:
    F = 1.8*eval(TempStr[0:-1]) + 32
    print("转换后的温度是{:.2f}F".format(F))
else:
    print("输入格式错误")

请输入带有符号的温度值：1F
转换后的温度是-17.22C


## 注释
* 单行注释
使用 ‘#’后面跟注释信息

* 使用两个‘'''’ 包裹起来

In [8]:
# 这个是单行注释

'''
这个是多行注释
第二行
第三行
'''

'\n这个是多行注释\n第二行\n第三行\n'

## 保留字
基本单词
|保留字|作用|
|--|--|
|and||
|as||
|assert||
|break||
|class||
|continue||
|def||
|elif||
|else||
|except||
|finally||
|for||
|from||
|if||
|import||
|in||
|is||
|lambda||
|not||
|or||
|pass||
|raise||
|return||
|try||
|while||
|with||
|yield||
|del||
|global||
|nonlocal||
|True||
|Flase||
|None||

In [18]:
tempStr = input()
if (tempStr[-1] in ['F','f']):
    print("{:.2f}C".format((eval(tempStr[0:-1]) - 32) / 1.8))
elif (tempStr[-1] in ['C','c']):
    print("{:.2f}F".format( eval(tempStr[0:-1]) * 1.8 + 32))
else:
    print('输入格式错误')

54.21F
12.34C


## 数字形式转换 I
获得用户输入的一个正整数输入，输出该数字对应的中文字符表示。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

0到9对应的中文字符分别是：零一二三四五六七八九


```
输入示例1‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬
123
输出示例1‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬
一二三
```


```
输入示例2‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬
9876543210
输出示例2‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬
九八七六五四三二一零
```

> print()中增加end=""参数表示输出后不增加换行，多个print()可以连续输出。

In [28]:
str = "零一二三四五六七八九"
showStr = input()
show = ''
for i in showStr :
    show += str[eval(i)]
print(show)

123
一二三


In [29]:
template = "零一二三四五六七八九"

s = input()
for c in s:
    print(template[eval(c)], end="")

123
一二三

## 温度转换2

温度的刻画有两个不同体系：摄氏度（Celsius）和华氏度（Fahrenheit）。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

请编写程序将用户输入华氏度转换为摄氏度，或将输入的摄氏度转换为华氏度。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

转换算法如下：（C表示摄氏度、F表示华氏度）‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

C = ( F - 32 ) / 1.8‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

F = C * 1.8 + 32‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

要求如下：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

(1) 输入输出的摄氏度采用大写字母C开头，温度可以是整数或小数，如：C12.34指摄氏度12.34度；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

(2) 输入输出的华氏度采用大写字母F开头，温度可以是整数或小数，如：F87.65指华氏度87.65度；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

(3) 不考虑异常输入的问题，输出保留小数点后两位；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

(4) 使用input()获得测试用例输入时，不要增加提示字符串。

In [32]:
tempStr = input()
if (tempStr[0] in ['C','c']) :
    print("F{:.2f}".format( eval(tempStr[1:]) * 1.8 + 32))
else:
    print("C{:.2f}".format((eval(tempStr[1:]) - 32) / 1.8))

C12.34
F54.21


## 货币转换 I

人民币和美元是世界上通用的两种货币之一，写一个程序进行货币间币值转换，其中：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

人民币和美元间汇率固定为：1美元 = 6.78人民币。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

程序可以接受人民币或美元输入，转换为美元或人民币输出。人民币采用RMB表示，美元USD表示，符号和数值之间没有空格。‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

注意：‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

(1) 这是一个OJ题目，获得输入请使用input() ；‪‬‪‬‪‬‪‬‪‬‮‬‪‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‫‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‮‬‭‬‫‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‪‬‪‬‪‬‪‬‪‬‪‬‮‬‪‬‭‬

(2) 不提示输出格式错误，结果小数点后保留两位。

In [37]:
str = input()
if (str[0:3] == 'USD') :
    print('RMB{:.2f}'.format(eval(str[3:]) * 6.78))
else:
    print('USD{:.2f}'.format(eval(str[3:]) / 6.78))

RMB123
USD18.14


# 绘制蟒蛇

引入 turtle 库 也叫海龟库

In [45]:
import turtle
turtle.setup(650, 350, 200, 200)
turtle.penup()
turtle.fd(-250)
turtle.pendown()
turtle.pensize(25)
turtle.pencolor("green")
turtle.seth(-40)
for i in range(4) :
    turtle.circle(40,80)
    turtle.circle(-40,80)
turtle.circle(40, 80/2)
turtle.fd(40)
turtle.circle(16,180)
turtle.fd(40* 2/3)
turtle.done()

## 库引用
### 基础语法
 import <库名>
 
 > 不会出现函数重名问题
 
 ### 使用不用带库名
 使用 from和import保留字完成
 from <库名> import <函数名>
 form <库名> import *

In [23]:
from turtle import *
setup(650, 350, 200, 200)
circle(40,360)
fd(100)
circle(40,-360)
done()

自己绘制一个蟒蛇

In [35]:
import turtle
turtle.setup(500,500,100,100)
turtle.penup()
turtle.bk(200)
turtle.pendown()
turtle.pensize(20)
turtle.pencolor("green")
turtle.seth(-40)
for i in range(4):
    turtle.circle(30,80)
    turtle.circle(-30,80)
turtle.circle(30,40)
turtle.fd(50)
turtle.circle(16,180)
turtle.fd(50* 1/3)
turtle.done()


画一个正方形

In [51]:
import turtle
turtle.setup(500,500,100,100)
turtle.fd(100)
turtle.seth(90)
turtle.fd(100)
turtle.seth(180)
turtle.fd(100)
turtle.seth(-90)
turtle.fd(100)
turtle.seth(0)
turtle.done()

绘制六边形

In [59]:
import turtle
turtle.setup(500,500,100,100)
# 1
turtle.seth(0)
turtle.fd(100)
# 2
turtle.seth(60)
turtle.fd(100)
# 3
turtle.seth(120)
turtle.fd(100)
# 4
turtle.seth(180)
turtle.fd(100)
# 5
turtle.seth(-120)
turtle.fd(100)
# 6
turtle.seth(-60)
turtle.fd(100)
turtle.seth(0)
turtle.done()

In [63]:
import turtle
turtle.setup(500,500,100,100)
for i in range(6):
    turtle.fd(100)
    turtle.left(60)
turtle.seth(0)
turtle.done()

In [1]:
import turtle as t
t.pensize(2)
for i in range(9):
    t.fd(160)
    t.left(80)  #720/9
t.done()

In [21]:
import turtle as t
t.pensize(2)
for i in range(4):
    t.seth(90*i)
    t.fd(150)
    t.right(90)
    t.circle(-150, 45)
    t.goto(0,0)
t.done()

In [26]:
str = "Hello World"
num = input()
if eval(num) == 0:
    print(str)
elif eval(num) > 0:
    for i in range(0, len(str), 2):
        print(str[i],end="")
        if (i < len(str) - 1) :
            print(str[i+1])
else:
    for i in range(len(str)):
        print(str[i])

1
He
ll
o 
Wo
rl
d

In [37]:
str = input()
# 标记第几个个处理
tag = 0
m = ''
op = ''
n = ''
for i in range(len(str)):
    if str[i] != " ":
        if tag == 0 :
            m += str[i]
        elif tag == 1 :
            op += str[i]
        else: 
            n += str[i]
    else :
        tag+=1
print("{:.2f}".format(eval(m + op + n)))

10 + 100
110.00
